cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/KubeTwillRunnerService.java

```diff
@@ -194,6 +197,27 @@ public KubeTwillRunnerService(MasterEnvironmentContext masterEnvContext, ApiClie
     this.workloadIdentityProvider = workloadIdentityProvider;
   }
 
+  /**
+   * Returns true if the pod is owned by a stateful set.
+   */
+  private boolean isStatefulSetPod() {
+    return this.podInfo.getOwnerReferences().size() == 1
```

Comment:
can owner reference be >1?

Reply:
Yes, according to this article - https://stackoverflow.com/questions/59759438/when-will-a-kube-object-have-more-than-one-ownerreferences , it is a possibility. But it applies for a very general case.
The owner reference in case of AppFabric pod and Preview pods will point to the single stateful set that created them.

Since it is a list, I am adding this check before doing a get(0).

Reply:
Agree, but calling isStatefulSetPod in future on a sts pod with multiple owner reference will return false which is wrong. Why not checking >=1 ?

Reply:
I have modified this function to - 


`
    return this.podInfo.getOwnerReferences().stream().anyMatch(r -> r.getKind().equals("StatefulSet"));

`